### PR TITLE
Added jansson and cpuminer

### PR DESCRIPTION
**Pull request by demhademha**

### CODE REVIEW

**Review by a0ahmad — APPROVED** [review]

File: jansson.mk [hunk 1]
```diff
@@ -17,8 +17,8 @@ jansson:
 else
 jansson: jansson-setup
 	cd $(BUILD_WORK)/jansson && ./configure \
-			--host=$(GNU_HOST_TRIPLE) \
-	--prefix=/usr 
+			--host=$(GNU_HOST_TRIPPLE) \
```
Comment (CRKatri) [inline jansson.mk:20]: `GNU_HOST_TRIPLE`, not `GNU_HOST_TRIPPLE`

File: jansson.mk [hunk 2]
```diff
@@ -17,8 +17,8 @@ jansson:
...
+	--prefix=/usr
```
Comment (CRKatri) [inline jansson.mk:21]: Nothing serious, but this indention needs fixing.

File: jansson.mk [hunk 3]
```diff
@@ -27,21 +27,32 @@ jansson: jansson-setup
 	touch $(BUILD_WORK)/jansson/.build_complete
 endif
 
+
 jansson-package: jansson-stage
 	# jansson.mk Package Structure
-	rm -rf $(BUILD_DIST)/jansson
-	mkdir -p $(BUILD_DIST)/jansson
+	rm -rf $(BUILD_DIST)/*jansson*/
```
Comment (CRKatri) [inline jansson.mk:33]: Don't use wildcards here, it'll remove the deb.

File: jansson.mk [hunk 4]
```diff
@@ -27,21 +27,32 @@ jansson: jansson-setup
...
 	$(call SIGN,jansson,general.xml)
 	
 	# jansson.mk Make .debs
 	$(call PACK,jansson,DEB_JANSSON_V)
+	$(call PACK,libjansson-dev,DEB_JANSSON_V)
+	$(call PACK,libjansson4,DEB_JANSSON_V)
 	
 	# jansson.mk Build cleanup
-	rm -rf $(BUILD_DIST)/jansson
+	rm -rf $(BUILD_DIST)/*jansson*/
```
Comment (CRKatri) [inline jansson.mk:56]: Don't use wildcards here, it'll remove the deb.

File: jansson.mk [hunk 5]
```diff
@@ -27,21 +27,32 @@ jansson: jansson-setup
...
+	cp -a $(BUILD_STAGE)/jansson/usr/lib/libjansson.la $(BUILD_DIST)/libjansson-dev/usr/lib
+	cp -a $(BUILD_STAGE)/jansson/usr/lib/pkgconfig $(BUILD_DIST)/libjansson-dev/usr/lib
 	
-	# jansson.mk Prep jansson
-	cp -a $(BUILD_STAGE)/jansson/usr $(BUILD_DIST)/jansson
+	# jansson.mk Prep libjansson4
+	cp -a $(BUILD_STAGE)/jansson/usr/lib/libjansson*.d* $(BUILD_DIST)/libjansson4/usr/lib
 	
 	# jansson.mk Sign
 	$(call SIGN,jansson,general.xml)
```
Comment (CRKatri) [inline jansson.mk:48]: You need to sign the libraries

File: jansson.mk [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	+$(MAKE) -C $(BUILD_WORK)/jansson install \
+		DESTDIR="$(BUILD_BASE)"
+	touch $(BUILD_WORK)/jansson/.build_complete
+endif
+
+
+jansson-package: jansson-stage
+	# jansson.mk Package Structure
+	rm -rf $(BUILD_DIST)/*jansson*/
+	mkdir -p $(BUILD_DIST)/jansson/usr \
```
Comment (CRKatri) [inline jansson.mk:34]: Unnecessarily making `$(BUILD_DIST)/jansson/usr/` which doesnt get used.

File: build_info/jansson.control [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Package: jansson
```
Comment (CRKatri) [inline build_info/jansson.control:1]: This control files never gets used, it can be removed.

File: build_info/cpuminer.control [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Package: cpuminer
+Version: @DEB_CPUMINER_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
+Section: Utilities 
+Priority: optional
+Description: A multi-threaded CPU miner for Litecoin and Bitcoin
+Depends: jansson, curl
```
Comment (CRKatri) [inline build_info/cpuminer.control:8]: This should depend on `libjansson4`

File: build_info/libjansson-dev.control [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Package: libjansson-dev
+Version: @DEB_JANSSON_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
+Section: Libraries
+Priority: optional
+Description: A C library for encoding, decoding and manipulating JSON 
+Tag: purpose::console, role::developer
```
Comment (CRKatri) [inline build_info/libjansson-dev.control:8]: Add, `Depends: libjansson4 (= @DEB_JANSSON_V@)`

File: build_info/libjansson-dev.control [hunk 10]
```diff
@@ -0,0 +1,8 @@
+Package: libjansson-dev
+Version: @DEB_JANSSON_V@
+Architecture: @DEB_ARCH@
+Maintainer: @DEB_MAINTAINER@
+Section: Libraries
```
Comment (CRKatri) [inline build_info/libjansson-dev.control:5]: Should go in the `Development` section.

File: jansson.mk [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+	cp -a $(BUILD_STAGE)/jansson/usr/lib/pkgconfig $(BUILD_DIST)/libjansson-dev/usr/lib
+	
+	# jansson.mk Prep libjansson4
+	cp -a $(BUILD_STAGE)/jansson/usr/lib/libjansson*.d* $(BUILD_DIST)/libjansson4/usr/lib
+	
+	# jansson.mk Sign
+	$(call SIGN,jansson,general.xml)
+	
+	# jansson.mk Make .debs
+	$(call PACK,jansson,DEB_JANSSON_V)
```
Comment (CRKatri) [inline jansson.mk:51]: This can be removed.

**Review by CRKatri — CHANGES_REQUESTED** [review]

**Review by a0ahmad — APPROVED** [review]

**Comment by demhademha** [comment]
@CRKatri should be fixed now.

**Review by Diatrus — APPROVED** [review]
Thank you very much for the contribution. It means a lot!

**Pull request merged**